_posts/2024-04-30-A-comprehensive-guide-to-setup-and-connect-self-managed-dashboards-with-Amazon-OpenSearch-domain.md

```diff
@@ -0,0 +1,276 @@
+```yml
+version: '3'
+services:
+  opensearch-dashboards:
+    image: opensearch-dashboards-no-security
+    container_name: opensearch-dashboards
+    ports:
+      - 5601:5601
+    expose:
+      - "5601"
+    environment:
+      OPENSEARCH_HOSTS: '["https://success-2-ce6hkjt5gh.ap-south-1.es.amazonaws.com"]'
+    networks:
+      - opensearch-net
+networks:
+  opensearch-net:
+```
+# 
+# Guide to setup self-managed dashboards in ECS & in EC2 hosted docker container - HTTP basic authentication
+
+## Prerequisite
+AWS managed OpenSearch domain must incorporate Fine-Grained Access Control (FGAC) with HTTP basic authentication, ensuring that a master user is created in the internal user database.
+[Reference here](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/fgac-http-auth.html)
+
+## Steps to spin up a self-managed dashboards in ECS
+1. Create a task within ECS Fargate using the dashboards docker image.
+2. When creating a task, under "container definition" in "port mapping" make sure the container ports `5601` and `9200` are added.
+3. Under environment variables, add the mandatory keys and values mentioned in this [doc](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/dashboards.html#dashboards-local) to seamlessly connect with managed service domain.
+
+> [!NOTE]
+> Below is the sample `task.json` file from the ECS task defenition to help understand the environment variables that has to be set while creating tasks
+```json
+{
+...
+            "portMappings": [
+                {
+                    "name": "dash-5601-tcp",
+                    "containerPort": 5601,
+                    "hostPort": 5601,
+                    "protocol": "tcp",
+                    "appProtocol": "http"
+                },
+                {
+                    "name": "dash-9200-tcp",
+                    "containerPort": 9200,
+                    "hostPort": 9200,
+                    "protocol": "tcp",
+                    "appProtocol": "http"
+                }
+            ],
+            "essential": true,
+            "environment": [
+                {
+                    "name": "OPENSEARCH_USERNAME",
+                    "value": "xxx"
+                },
+                {
+                    "name": "OPENSEARCH_HOSTS",
+                    "value": "https://success-2-ce6hkjt5gh.ap-south-1.es.amazonaws.com"
+                },
+                {
+                    "name": "OPENSEARCH_PASSWORD",
+                    "value": "xxxx"
+                }
+            ],
+...
+}
+```
+
+4. Create a service using the previously created task within the identical VPC and subnet where the managed service OpenSearch domain is operating.
+5. Access the self-managed dashboards by hitting the public endpoint of the running task in ECS Fargate. By doing so, you can conveniently view and interact with all the saved objects in accordance with the Fine-Grained Access Control settings.
```

Comment:
Please use a different/more precise word than "hitting".

Reply:
connecting to the EC2 endpoint